src/application/wizard/service.ts

```diff
@@ -29,10 +36,77 @@ export class WizardService {
             this.teamService.create({
                 role: 'author',
                 teamMembers,
-                objectId: id.value,
+                objectId: submissionId.value,
                 objectType: 'manuscript',
             });
         }
         return submission;
     }
+
+    async saveManuscriptFile(
+        submissionId: SubmissionId,
+        userId: string,
+        file: FileUpload,
+        fileSize: number,
+    ): Promise<Submission> {
+        const submission = await this.submissionService.get(submissionId);
+        this.checkOwnership(submission, userId);
+
+        const { filename, mimetype: mimeType, createReadStream } = await file;
+        const stream = createReadStream();
+
+        const manuscriptFile = await this.fileService.create(
+            submissionId,
+            filename,
+            mimeType,
+            fileSize,
+            FileType.MANUSCRIPT_SOURCE_PENDING,
+        );
+
+        const fileContents: Buffer = await new Promise((resolve, reject) => {
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            const chunks: Array<any> = [];
+            stream.on('data', chunk => chunks.push(chunk));
+            stream.on('error', reject);
+            stream.on('end', () => resolve(Buffer.concat(chunks)));
+        });
+
+        const uploadPromise = this.fileService.upload(fileContents, manuscriptFile);
+
+        manuscriptFile.setStatusToStored();
+
+        const semanticExtractionPromise = this.semanticExtractionService.extractTitle(
+            fileContents,
+            mimeType,
+            filename,
+            submissionId,
+        );
+
+        try {
+            await Promise.all([uploadPromise, semanticExtractionPromise]);
+        } catch (e) {
+            manuscriptFile.setStatusToCancelled();
```

Comment:
This makes sense on uploadPromise failure but not sure we'd want to cancel on extraction failure

Reply:
the semantic extraction doesn't throw, error gets caught and adds something to the log

Reply:
Some logging will be needed here on failed upload